### ISSUE TITLE
Baidu API with GeocoderAuthenticationFailure

### ISSUE THREAD

**Issue by vba34520** [geopy/geopy#394]
The version of geopy is 1.21.0.

```python
from geopy.geocoders import Baidu

geolocator = Baidu(api_key='xxx')
location = geolocator.geocode('天安门')
```

When I call Baidu API, it will get GeocoderAuthenticationFailure.

I degugged it and found geopy still using the v2 of Baidu when the version of Baidu is v3 now.

**Comment by vba34520** [comment]
modify the file `geopy\geocoders\baidu.py`
```python
api_path = '/geocoding/v3'
```

**Comment by KostyaEsmukov** [comment]
v2 seems to continue to work at the moment, so it looks like it was an intermittent problem.

And the key I have doesn't seem to work with v3, I'm getting an `Authentication Failure` error.

I cannot find any information on the differences between the versions, and why was it bumped. Looking at the docs (they seem to be only in Chinese, so I'm using a translator to English) I don't see any difference from v2, and the "Update log" (translated) tab doesn't contain any meaningful information.

If keys from v2 are no longer valid for v3, then I cannot just change the url in the next geopy version -- this would be a breaking change. I'd have to create a new geocoder class for v3.

Would be glad if anyone could bring some light on what is going on with this version bump and authentication keys.

**Comment by KostyaEsmukov** [comment]
Spent 2 hours trying to login to my Baidu account. For some reason they have reset my password and didn't let me to restore it neither with my email nor with my phone. The only thing that helped is trying to restore by username (I didn't know I had one!), which I saw as taken only trying to register a new account (in despair).

Anyway, I got to the API keys console and managed to create new API keys, and here are my conclusions:

- old keys definitely don't work with v3
- new keys definitely don't work with v2
- In v3 they have split geocoding from reverse geocoding to two different endpoints.

Fixed in f612130 by adding a new `BaiduV3` geocoder.
